modules/javafx.web/src/main/native/Source/WebCore/bindings/java/EventListenerManager.cpp

```diff
@@ -35,80 +35,66 @@ EventListenerManager& EventListenerManager::get_instance()
     return sharedManager;
 }
 
-void EventListenerManager::registerListener(JavaEventListener *ptr, const JLObject &listener)
+void EventListenerManager::registerListener(JavaEventListener *listener, const JLObject &listenerObj)
 {
-    JavaObjectWrapperHandler *temp_ref = new JavaObjectWrapperHandler(listener);
-    std::pair<JavaEventListener*, JavaObjectWrapperHandler*> entry{ ptr, temp_ref };
-    listener_lists.insert(entry);
+    ListenerJObjectWrapper *temp_ref = new ListenerJObjectWrapper(listenerObj);
+    std::pair<JavaEventListener*, ListenerJObjectWrapper*> entry{ listener, temp_ref };
+    listenerJObjectMap.insert(entry);
 }
 
-void EventListenerManager::unregisterListener(JavaEventListener *ptr)
+void EventListenerManager::unregisterListener(JavaEventListener *listener)
 {
-     std::map<JavaEventListener*, JavaObjectWrapperHandler*>::iterator it;
-     it = listener_lists.find(ptr);
+     std::map<JavaEventListener*, ListenerJObjectWrapper*>::iterator it;
+     it = listenerJObjectMap.find(listener);
 
-     if (it != listener_lists.end()) {
+     if (it != listenerJObjectMap.end()) {
          if (it->second && it->second->use_count() == 1) {
              delete it->second;
              it->second = nullptr;
-             listener_lists.erase(it); // remove from list
+             listenerJObjectMap.erase(it); // remove from list
          }
 
          if (it->second && it->second->use_count() > 1)
```

Comment:
Shouldn't this be `else if`? The previous block calls `erase(it)`, so it isn't valid to dereference `it` after that call.

Reply:
Applied and  tested